### ISSUE TITLE
Sign source tarball 

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-build#540]
Provide detached signature with zimlib and kiwix tarballs. Debian project now encouranges upstream to provided signed tarballs so we can make sure nothing is tampered with tarball while we download
https://www.gnupg.org/gph/en/manual/x135.html

**Comment by dilankavishka** [comment]
Can i work on this issue please? @kelson42

**Comment by dev-KartikSharma** [comment]
Hi @kelson42 , I noticed this issue was raised a year ago, and I wanted to check if it’s still available to work on. Is the assignment of this issue still open? I'd be happy to contribute if needed.

Thanks!

**Comment by kelson42** [comment]
@dev-KartikSharma Yes, how do you want to proceed?

**Comment by dev-KartikSharma** [comment]
to provide a detached signature for your tarballs using zimlib and kiwix can use GPG to generate and verify the signature of tarballs thats the idea i'm thinking of

**Comment by dev-KartikSharma** [comment]
Hi @kelson42,
I’ve made a few changes to builder.py to handle the GPG tarballs signature. I believe it’s ready for a PR, but I wanted to confirm if these changes align with the project's goals. Looking forward to your feedback before proceeding!
Thanks in advance!

**Comment by kelson42** [comment]
@dev-KartikSharma Please make PR so we can have a look please

**Comment by dev-KartikSharma** [comment]
I’ve submitted the PR please take a look when you get a chance. I’m waiting for your review or any suggestions for further changes.

**Comment by dev-KartikSharma** [comment]
@kelson42 was I supposed to mention for review in the Pull request??

**Comment by kelson42** [comment]
@dev-KartikSharma yes and no... but now this is clear that your PR is ready to review.

**Comment by dev-KartikSharma** [comment]
@kelson42 it's was the first ever pr so I had no idea about it😅

**Comment by Rancho-rachit** [comment]
Which path should be followed to solve this issue, such that generating gpg detached signature file to verify the authenticity of source tarballs

Path 1: <ins>[Explicitly]</ins> Use GitHub scripts & workflow to generate the signature file after creating archives.
Path 2: <ins>[Implicitly]</ins> Modify the core build system kiwix-build to generate the detached signature file.
